module-presentation/src/main/java/com/depromeet/memory/service/TimelineServiceImpl.java

```diff
@@ -0,0 +1,151 @@
+package com.depromeet.memory.service;
+
+import com.depromeet.image.Image;
+import com.depromeet.image.dto.response.MemoryImagesDto;
+import com.depromeet.memory.Memory;
+import com.depromeet.memory.Stroke;
+import com.depromeet.memory.dto.response.StrokeResponseDto;
+import com.depromeet.memory.dto.response.TimelineResponseDto;
+import com.depromeet.memory.repository.MemoryRepository;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Slice;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Slf4j
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class TimelineServiceImpl implements TimelineService {
+    private final MemoryRepository memoryRepository;
+
+    @Override
+    public Slice<TimelineResponseDto> getTimelineByMemberIdAndCursor(
+            Long memberId, Long cursorId, String recordAt, int pageSize) {
+        LocalDate recordAtLocalDate = null;
+        if (recordAt != null && !recordAt.trim().isEmpty()) {
+            recordAtLocalDate =
+                    LocalDate.parse(recordAt, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+        }
+
+        Pageable pageable = PageRequest.of(0, pageSize, Sort.by(Sort.Order.desc("recordAt")));
+        Slice<Memory> memories =
+                memoryRepository.getSliceMemoryByMemberIdAndCursorId(
+                        memberId, cursorId, recordAtLocalDate, pageable);
+        return memories.map(this::mapToTimelineResponseDto);
+    }
+
+    public TimelineResponseDto mapToTimelineResponseDto(Memory memory) {
+        return TimelineResponseDto.builder()
+                .memoryId(memory.getId())
+                .recordAt(memory.getRecordAt().toString())
+                .startTime(localTimeToString(memory.getStartTime()))
+                .endTime(localTimeToString(memory.getEndTime()))
+                .lane(memory.getLane())
+                .diary(memory.getDiary())
+                .totalMeter(calculateTotalMeter(memory.getStrokes(), memory.getLane()))
+                .memoryDetailId(
+                        memory.getMemoryDetail().getId() == null
+                                ? null
+                                : memory.getMemoryDetail().getId())
```

Comment:
이거를 저는 null이 아닌 걸 먼저 써서

```suggestion
                        memory.getMemoryDetail().getId() != null
                                ? memory.getMemoryDetail().getId())
                                : null
```

이렇게 적었는데 통일하는 게 어떨까요! 정해주시면 제가 맞춰도 좋구 성주님이 수정해 주셔도 좋구요 !!

Reply:
오 요건 어떻게 한거야?? 신기방기

Reply:
귀엽징 ~~ 요거 누르면 돼 므히힝
<img width="370" alt="Screenshot 2024-07-19 at 01 42 00" src="https://github.com/user-attachments/assets/925d803c-375f-4860-bfa1-e1e5740f423d">

Reply:
수정 완료